### PR TITLE
Clean ci/validate_asciidoc.sh

### PR DESCRIPTION
**Pull request by marco-antognini-sonarsource**
This PR is a prerequisite for:
 * #2685

---

 * Regroup readonly variable declaration. `declare -xr` is used for exported readonly variable.
 * Use more accurate log messages.
 * Use consistent indentation of 2 spaces.
 * Fail fast if dependencies cannot be installed.
 * Regroup HTML generation & its validation.
 * Document the main validation points.
 * Simplify some commands and address some ShellCheck warnings.

### CODE REVIEW

File: ci/validate_asciidoc.sh [hunk 1]
```diff
@@ -1,11 +1,26 @@
 #!/bin/bash
 set -uo pipefail
 
+# Install script dependencies
+(set -e; cd rspec-tools && pipenv install)
```
Comment (necto) [inline ci/validate_asciidoc.sh:5]: IIUC, `set -e` will only terminate the subshell on an error. If you are going for the early failure of the entire script, you'd have to do something like
```suggestion
set -e
cd rspec-tools && pipenv install
set +e
```

File: ci/validate_asciidoc.sh [hunk 2]
```diff
@@ -1,11 +1,26 @@
...
+
+# This script runs all tests; it doesn't exit at the first failure.
+exit_code=0
+
 readonly ALLOWED_RULE_SUB_FOLDERS=['common'];
+readonly ROOT=$PWD
+declare -xr PATH_WITH_VARIABLE="$(realpath ./ci/replace_variables_in_path.sh)"
```
Comment (necto) [inline ci/validate_asciidoc.sh:12]: As I had to google and read up on `declare -xr` and I expect many people who have to touch this file are also not fluent in Bash enough to understand it, I would add a comment explaining what this line does (for my selfish convenience).

```suggestion
# Make constant PATH_WITH_VARIABLE available to child processes
declare -xr PATH_WITH_VARIABLE="$(realpath ./ci/replace_variables_in_path.sh)"
```

**Review by necto — APPROVED** [review]
Thanks for overhauling the script!

File: ci/validate_asciidoc.sh (hunk 1 above)
Comment (marco-antognini-sonarsource) [reply ci/validate_asciidoc.sh:5]: Indeed, I made a mistake while testing! Thanks for catching this.

File: ci/validate_asciidoc.sh [hunk 3]
```diff
@@ -47,37 +54,34 @@ do
...
-        exit_code=1
+      echo "ERROR: These adoc files contain an include that is stuck to other content."
+      echo "This may result in broken tags and other display issues."
+      echo "Make sure there is an empty line before and after each include:"
+      cat stuck
+      exit_code=1
     fi
     rm -f stuck
 
     supportedLanguages=$(sed 's/ or//' supported_languages.adoc | tr -d '`,')
```
[the remark this answers is not in the document]
Comment (marco-antognini-sonarsource) [reply ci/validate_asciidoc.sh]: Done in d27ef335f9f70685e920f7218f9769dd4075ed3e

**Pull request merged**